File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/services/CountryService.java
package com.chaincuerealestate.chaincuerealestate.services;

import com.chaincuerealestate.chaincuerealestate.domains.Country;
import com.chaincuerealestate.chaincuerealestate.repositories.CountryRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.function.BiConsumer;
import java.util.function.Function;

@Service
@RequiredArgsConstructor
public class CountryService implements CountryServiceI {
    private final CountryRepository countryRepository;

    @Override
    public Country save(Country.CountryNames countryNames) {
        var country = Country.create(countryNames);
        return countryRepository.save(country);
    }

    @Override
    public List<Country> findAll() {
        return countryRepository.findAll();
    }

}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/services/CountryService.kt
package com.example.chaincuerealestate.services

import com.example.chaincuerealestate.domains.Country
import com.example.chaincuerealestate.repositories.CountryRepository
import org.springframework.stereotype.Service

@Service
class CountryService(private val countryRepository: CountryRepository) : CountryServiceI {

    override fun save(countryNames: Country.CountryNames): Country {
        val country = Country.create(countryNames)
        return countryRepository.save(country)
    }

    override fun findAll(): List<Country> {
        return countryRepository.findAll()
    }
}

